tests/test_charm.py

```diff
@@ -60,12 +64,83 @@
     "regex": "(.*)",
 }
 
-DASHBOARD_DUMMY_DATA_1 = {"request_12345678": json.dumps({"dashboard": "dummy_data_1"})}
+DASHBOARD_DUMMY_DATA_1 = {
+    "request_12345678": json.dumps(
+        {
+            "dashboard": {
+                "dashboard": {
+                    "__inputs": [
+                        {"pluginName": "Prometheus"},
+                    ],
+                    "templating": {
+                        "list": [
+                            {"datasource": "Juju data"},
+                        ],
+                    },
+                    "panels": {"data": "some_data_to_hash_across"},
+                },
+            },
+        }
+    )
+}
+
+DUMMY_FIXED_1 = {
+    "charm": "dashboard-app-1",
+    "content": '{"__inputs": [], "templating": {"list": [{"datasource": '
+    '"${prometheusds}"}]}, "panels": {"data": '
+    '"some_data_to_hash_across"}}',
+    "juju_topology": {
+        "application": "dashboard-app-1",
+        "model": "testmodel",
+        "model_uuid": "1234567890",
+        "unit": "dashboard-app-1/0",
+    },
+}
+
 DASHBOARD_DUMMY_DATA_2 = {
-    "request_87654321": json.dumps({"dashboard": "different_enough_to_rehash"})
+    "request_87654321": json.dumps(
+        {
+            "dashboard": {
+                "dashboard": {
+                    "templating": {
+                        "list": [
+                            {"name": "host"},
+                        ],
+                    },
+                    "panels": {"data": "different_enough_to_rehash"},
+                },
+            },
+        }
+    )
+}
+
+DUMMY_FIXED_2 = {
+    "charm": "dashboard-app-2",
+    "content": '{"templating": {"list": [{"allValue": null, "datasource": '
+    '"${prometheusds}", "definition": '
+    '"label_values(up{juju_model=\\"$juju_model\\",juju_model_uuid=\\"$juju_model_uuid\\",juju_application=\\"$juju_application\\"},host)", '
+    '"description": null, "error": null, "hide": 0, "includeAll": '
+    'false, "label": "hosts", "multi": true, "name": "host", '
+    '"options": [], "query": {"query": '
+    '"label_values(up{juju_model=\\"$juju_model\\",juju_model_uuid=\\"$juju_model_uuid\\",juju_application=\\"$juju_application\\"},host)", '
+    '"refId": "StandardVariableQuery"}, "refresh": 1, "regex": "", '
+    '"skipUrlSync": false, "sort": 1, "tagValuesQuery": "", "tags": '
+    '[], "tagsQuery": "", "type": "query", "useTags": false}]}, '
+    '"panels": {"data": "different_enough_to_rehash"}}',
+    "juju_topology": {
+        "application": "dashboard-app-2",
+        "model": "testmodel",
+        "model_uuid": "1234567890",
+        "unit": "dashboard-app-2/0",
+    },
```

Comment:
Are these hard coded structures of strings our only feasible option?

Reply:
Considering that they're isolated for test data only, and are predictable due to the patching, there's no reason at all to parameterize them unless you plan to re-use these test methods in some other testsuite.